src/utils.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Apr 16 19:12:51 2023
+
+@author: Sumedh Prasad
+
+utils code
+"""
+
+import logging
+import os
+from pathlib import Path
+from typing import List, Tuple
+
+import mysql.connector as mysql
+import pandas as pd
+from dotenv import load_dotenv
+import sys
+
+sys.dont_write_bytecode = True
+
+
+logging.basicConfig(filename='myapp.log',
+                    format='%(asctime)s %(message)s', 
+                    level=logging.INFO)
+
+
+# STEP 1
+def connect_db(host: str, user: str) -> Tuple[mysql.connection.MySQLConnection, str]:
+    """this function connects to mysql database and returns conn, cur as tuple
+
+    Args:
+        host (str): pass host name
+        user (str): pass user name
+
+    Returns:
+        Tuple[mysql.connection.MySQLConnection, str]: 
+    """
+    try:
+        logging.info('Started MySQl Connection Started')
+        conn = mysql.connect(host=host,
+                            user=user,
+                            password=os.getenv("PASSWORD"))
+        cur = conn.cursor()
+        logging.info("Connected to MySQL successfully!")
+        logging.info('Ended MySQl Connection Started')
+    except Exception as e:
+        logging.debug(e)
+    return conn, cur
+    
+
+# # STEP 2 CREATE A FUNCTION: GET_DATA OR READ_DATA
+# Filename argurment
+def read_data(filename: str) -> pd.DataFrame:
+    '''
+    Reads a CSV file and returns a pandas DataFrame.
+
+    Parameters
+    ----------
+    filename : str
+    - file_path (str): The path to the CSV file.   
+
+    Returns
+    -------
+    df : TYPE
+        DESCRIPTION.
+    - pandas.DataFrame: A DataFrame containing the data from the CSV file.
+
+    '''
+    try:
+        logging.info('Started  reading csv format file')
+        df = pd.read_csv(filename)
+        df.drop(columns = ['Unnamed: 0'], inplace = True, errors = 'ignore')
+        logging.info("Read CSV File successfully!")
+        logging.info('Ended reading csv format file')
+    except Exception as e:
+        logging.debug(e)
+        
+    return df
+
+
+# # STEP 3 CREATE FUNCTION CREATE DATABASE - RETURNS DATABASES
+# database name 
+def create_database(database_name: str, cur: str): 
+    '''
+    This function will create the database and will return list of all databases
+    
+    Parameters
+    ----------
+    database_name : str
+        Pass the Database name which we want to create it
+
+    Returns
+    -------
+    databases : TYPE
+        DESCRIPTION.
+
+    '''
+    try:
+        logging.info('Started  creating database')
+        cur.execute(f'DROP DATABASE IF EXISTS {database_name}')
+        cur.execute(f"CREATE DATABASE {database_name}")
+        cur.execute("SHOW DATABASES")
+        databases = cur.fetchall()
+        logging.info("Created Databse successfully!")
+        logging.info('Ended creating database')
+    except Exception as e:
+         logging.debug(e)
+         
+    return databases
```

Comment:
here you don't need to return databases. Instead you can just print databases before the except statement.